tests/basic_test.go

```diff
@@ -235,6 +296,28 @@ func TestDownloadFile(t *testing.T) {
 	if strings.Compare(text, expected) != 0 {
 		t.Errorf("Got different text than expected, expected %q got, %q", expected, text)
 	}
+
+	fileStat, err := os.Stat("./tmp/output.txt")
+	if err != nil {
+		t.Errorf("Result file could not be read: %s", err)
+	}
+
+	if fileStat.Size() != fileInfos.Size {
+		t.Errorf("File size does not match")
+	}
+
+	if fileInfos.Mtime == 0 {
+		t.Errorf("No file mtime preserved")
+	}
+
+	if fileInfos.Atime == 0 {
+		t.Errorf("No file atime preserved")
+	}
+
+	if fileInfos.Permissions == 0 {
+		t.Errorf("No file permissions preserved")
+	}
+
```

Comment:
What happens when `atime` is parsed in the wrong way so that it is differnet from zero but not the expected value? I would expect a comparison between the received value and the actual value here, what that be difficult to add?